html exports that depend on javascript fetched from cdn no longer render

I have some jupyter notebook widgets built with ipydatagrid which used to render nicely when using the notebook File->Export as->HTML

Now, these exported notebooks no longer render properly. Even .html files that used to work, no longer render.

Looking in firefox's javascript console, I see the browser starts by complaining:
```
Loading failed for the <script> with source “file:///home/zzz/Downloads/ipydatagrid.js”.
```
(which I believe is expected) and then it says:
```
Falling back to https://cdn/jsdelivr.net/npm/ for ipydatagrid^1.1.16
https://cdn.jsdelivr.net/npm/ipydatagrid@^1.1.16/dist/index.js
```
So far so good, I think, but then:
```
Error: Failed to deserialize data
    e https://cdn.jsdelivr.net/npm/ipydatagrid@^1.1.16/dist/index.js:2
    deserialize https://cdn.jsdelivr.net/npm/ipydatagrid@^1.1.16/dist/index.js:2
    _deserialize_state https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:2
    _make_model https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:16
    new_model https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:16
    set_state https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:16
    set_state https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:16
    promise callback*set_state https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:16
    I https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:26
    I https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:26
    I https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:26
    I https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:26
    promise callback*I https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:26
    <anonymous> https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:29
    execCb https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:29
    check https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:18
    enable https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:23
    init https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:17
    g https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:26
    setTimeout handler*h.nextTick< https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:32
    g https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:26
    requirejs https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:31
    <anonymous> https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:29
    <anonymous> https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:29
[embed-amd.js:16:1002392](https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js)
    _make_model https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:16
    new_model https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:16
    set_state https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:16
    set_state https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:16
    (Async: promise callback)
    set_state https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:16
    I https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:26
    I https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:26
    I https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:26
    I https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:26
    (Async: promise callback)
    I https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:26
    <anonymous> https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:29
    execCb https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:29
    check https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:18
    enable https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:23
    init https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:17
    g https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:26
    (Async: setTimeout handler)
    nextTick https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:32
    g https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:26
    requirejs https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js:31
    <anonymous> https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:29
    <anonymous> https://unpkg.com/@jupyter-widgets/html-manager@*/dist/embed-amd.js:29
```

The javascript file exists and can be downloaded, but doesn't seem to work.

Google Chrome gives similar errors:
```
embed-amd.js:16 Error: Failed to deserialize data
    at Object.e [as deserialize] (index.js:2:780772)
    at Object.deserialize (index.js:2:2504501)
    at v._deserialize_state (embed-amd.js:2:373454)
    at $n._make_model (embed-amd.js:16:1002558)
```

If instead I copy nbextensions/ipydatagrid/index.js to ipydatagrid.js and put it in the same directory as my exported html file, the browser finds the local file and the widgets render properly.

I'm using ipydatagrid 1.1.16, but have tried 1.2.1 and 1.3.0 as well with the same result.

Has something changed in the javascript that is distributed via cdn.jsdelivr.net ?

**Environment (please complete the following information):**
I'm using 
 - Gentoo linux, also tried on windows.
 - firefox 115.7 and chrome 121.0.6167.184

--- a/js/datagrid.ts
+++ b/js/datagrid.ts
@@ -83,6 +83,30 @@ function serialize_data(data: DataSource, manager: any): any {
 
 function deserialize_data(data: any, manager: any): DataSource {
   const deserialized_data: any = {};
+
+  // Backward compatibility for when data.data was an array of rows
+  // (should be removed in ipydatagrid 2.x?)
+  if (Array.isArray(data.data)) {
+    if (data.data.length === 0) {
+      return new DataSource(deserialized_data, data.fields, data.schema, true);
+    }
+
+    const unpacked = unpack_raw_data(data.data);
+    // Turn array of rows (old approach) into a dictionary of columns as arrays (new approach)
+    for (const column of Object.keys(unpacked[0])) {
+      const columnData = new Array(unpacked.length);
+      let rowIdx = 0;
+
+      for (const row of unpacked) {
+        columnData[rowIdx++] = row[column];
+      }
+
+      deserialized_data[column] = columnData;
+    }
+
+    return new DataSource(deserialized_data, data.fields, data.schema, true);
+  }
+
   for (const column of Object.keys(data.data)) {
     deserialized_data[column] = [];
 
@@ -467,7 +491,10 @@ export class DataGridView extends DOMWidgetView {
     });
 
     const grid_style = this.model.get('grid_style');
-    if (this.model.get('horizontal_stripes') || this.model.get('vertical_stripes')) {
+    if (
+      this.model.get('horizontal_stripes') ||
+      this.model.get('vertical_stripes')
+    ) {
       const index = this.model.get('horizontal_stripes')
         ? 'rowBackgroundColor'
         : 'columnBackgroundColor';

